lib/array_intersection.rb

```diff
@@ -1,3 +1,45 @@
+## Array Intersection
+
+# Design and implement a method that takes two integer arrays with unique values and returns their intersection in a new array.
+
+# For example:
+
+# ```
+# intersection([2, 3, 4], [4, 5, 6]) => [4]
+# intersection([50, 43, 25, 72], [25, 36, 43, 50, 80]) => [50, 25, 43]
+# intersection([9, 30, 42], [56, 34, 90, 32]) => []
+# ```
+
+
+
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  shortlist = []
+  longest = []
+  if list1.length < list2.length
+    shortlist = list1
+    longest = list2
+  else 
+    shortlist = list2
+    longest = list1
+  end
+  
+  hash = {}
+  shortlist.each do |item|
+  if 
+    longest.include?(item)
```

Comment:
`.include?` is an O(n) operation and since you have this in an O(m) loop... this becomes O(n * m)

Think about how you could use a hash to make this faster.